go/src/koding/socialapi/transport_test.go

```diff
@@ -0,0 +1,142 @@
+package socialapi_test
+
+import (
+	"encoding/json"
+	"log"
+	"net"
+	"net/http"
+	"strings"
+	"testing"
+
+	"koding/kites/tunnelproxy/discover/discovertest"
+	"koding/socialapi"
+)
+
+func TestTransport(t *testing.T) {
+	var storage TrxStorage
+	var auth = NewFakeAuth()
+
+	s, err := discovertest.NewServer(http.HandlerFunc(auth.GetSession))
+	if err != nil {
+		t.Fatalf("NewServer()=%s", err)
+	}
+	defer s.Close()
+
+	cache := socialapi.NewCache(auth.Auth)
+	cache.Storage = &storage
+
+	users := []*socialapi.Session{
+		{Username: "user1", Team: "foobar"},
+		{Username: "user2", Team: "foobar"},
+		{Username: "user3", Team: "foobar"},
+		{Username: "user", Team: "team"},
+	}
+
+	cases := []struct {
+		name  string
+		sess  *socialapi.Session // client
+		errs  []error            // fake errors for endpoint
+		codes []int              // fake response codes for endpoint
+		err   error              // final error from client
+		trx   TrxStorage         // underlying cache operations
+	}{{
+		"new user1",
+		users[0],
+		nil, nil, nil,
+		TrxStorage{
+			{Type: "get", Session: users[0]},
+			{Type: "set", Session: users[0]},
+		},
+	}, {
+		"cached user1",
+		users[0],
+		nil, nil, nil,
+		TrxStorage{
+			{Type: "get", Session: users[0]},
+		},
+	}, {
+		"new user2",
+		users[1],
+		nil, nil, nil,
+		TrxStorage{
+			{Type: "get", Session: users[1]},
+			{Type: "set", Session: users[1]},
+		},
+	}, {
+		"new user3 with an error",
+		users[2],
+		[]error{&net.DNSError{IsTemporary: true}}, nil, nil,
+		TrxStorage{
+			{Type: "get", Session: users[2]},
+			{Type: "set", Session: users[2]},
+		},
+	}, {
+		"cached user3 with an error and response codes",
+		users[2],
+		[]error{&net.DNSError{IsTemporary: true}}, []int{401, 401}, nil,
+		TrxStorage{
+			{Type: "get", Session: users[2]},
+		},
+	}}
+
+	rec := &AuthRecorder{
+		AuthFunc: cache.Auth,
+	}
+
+	client := http.Client{
+		Transport: &socialapi.Transport{
+			RoundTripper: &FakeTransport{},
```

Comment:
i believe we can use after a bit of refactoring. 

As we talked previously and all agreed on, instead of mocking lets use the real functionality in our tests.